packages/astro/src/vite-plugin-astro/hmr.ts

```diff
@@ -82,18 +82,17 @@ export async function handleHotUpdate(ctx: HmrContext, config: AstroConfig, logg
 		invalidateCompilation(config, file);
 	}
 
-	const mod = ctx.modules.find((m) => m.file === ctx.file);
+	// Bugfix: sometimes style URLs get normalized and end with `lang.css=`
+	// These will cause full reloads, so filter them out here
+	const mods = ctx.modules.filter(m => !m.url.endsWith('='));
```

Comment:
This was an edge case that was causing full reloads when they weren't called for. Sometimes `?astro&type=astro&index=0&lang.css` files would be duplicated in the module graph but represented as `?astro=&type=astro&index=0&lang.css=` which would trigger a full reload. We're removing these from the hot update handler since they're always irrelevant.